diff --git a/build/BUILD.gn b/build/BUILD.gn
index f3fff89..69c2003 100644
--- a/build/BUILD.gn
+++ b/build/BUILD.gn
@@ -37,10 +37,20 @@ if (mini_chromium_is_mac) {
     # win_sdk\bin\SetEnv.cmd inside this path will be used to configure the
     # Windows toolchain.
     win_toolchain_path = "<autodetect>"
+
+    # Link CRT dynamically or statically
+    dynamic_crt = false
   }
 }
 
 config("debug") {
+  if (mini_chromium_is_win) {
+    if (dynamic_crt) {
+      cflags = [ "/MDd" ]
+    } else {
+      cflags = [ "/MTd" ]
+    }
+  }
 }
 
 config("release") {
@@ -75,6 +85,12 @@ config("release") {
       "/LTCG",
     ]
     arflags = [ "/LTCG" ]
+
+    if (dynamic_crt) {
+      cflags += [ "/MD" ]
+    } else {
+      cflags += [ "/MT" ]
+    }
   }
 }
 
